### PR TITLE
chore: Bump SDK for composable bridging updates

### PR DESCRIPTION
**Pull request by pxrl**
This PR gets relayer-v2 over the hurdle of a new SDK bump. The behaviour implemented here should be unchanged - the main impact is to supply a Deposit object to the relayFeeCalculator getGasCosts() method, rather than a simple chain ID.

### CODE REVIEW

**Comment by pxrl** [comment]
Have applied `do not merge` since this is a pre-emptive PR; the SDK has not yet been bumped to 0.17.4 so it's not possible to merge this. Once the bump does occur, a sync of `yarn.lock` will be needed.

File: src/clients/ProfitClient.ts [hunk 1]
```diff
@@ -52,6 +63,14 @@ export const GAS_TOKEN_BY_CHAIN_ID: { [chainId: number]: string } = {
 // TODO: Make this dynamic once we support chains with gas tokens that have different decimals.
 const GAS_TOKEN_DECIMALS = 18;
 
+// @dev This address is known on each chain and has previously been used to simulate Deposit gas costs.
+// Since _some_ known recipient address is needed for simulating a fill, default to this one. nb. Since
+// the SpokePool implements custom behaviour when relayer === recipient, it's important not to use the
+// relayer's own address.
+// @todo: Consider using an address with confirmed 0 destinationToken balance so that the initial cost
+// of populating storage is included in the estimate.
+const testRecipient = "0xBb23Cd0210F878Ea4CcA50e9dC307fb0Ed65Cf6B";
```
Comment (james-a-morris) [inline src/clients/ProfitClient.ts:72]:
```suggestion
const TEST_RECIPIENT = "0xBb23Cd0210F878Ea4CcA50e9dC307fb0Ed65Cf6B";
```

File: src/clients/ProfitClient.ts [hunk 2]
```diff
@@ -97,6 +116,7 @@ export class ProfitClient {
     readonly hubPoolClient: HubPoolClient,
     spokePoolClients: SpokePoolClientsByChain,
     readonly enabledChainIds: number[],
+    readonly relayerAddress: string,
```
Comment (james-a-morris) [inline src/clients/ProfitClient.ts:119]: Good call to save this state here

File: src/clients/ProfitClient.ts [hunk 3]
```diff
@@ -139,18 +159,27 @@ export class ProfitClient {
...
 
   // @todo: Factor in the gas cost of submitting the RefundRequest on alt refund chains.
-  getTotalGasCost(chainId: number): BigNumber {
-    return this.totalGasCosts[chainId] ? toBN(this.totalGasCosts[chainId]) : ethersConstants.Zero;
+  getTotalGasCost(deposit: Deposit): BigNumber {
+    const { destinationChainId: chainId } = deposit;
+
+    // If there's no attached message, gas consumption from previous fills can be used in most cases.
+    // @todo: Simulate this per-token in future, because some ERC20s consume more gas.
+    if (resolveDepositMessage(deposit) === "0x" && isDefined(this.totalGasCosts[chainId])) {
```
Comment (james-a-morris) [inline src/clients/ProfitClient.ts:167]:
```suggestion
    if (isMessageEmpty(resolveDepositMessage(deposit)) && isDefined(this.totalGasCosts[chainId])) {
```

File: src/clients/ProfitClient.ts [hunk 4]
```diff
@@ -139,18 +159,27 @@ export class ProfitClient {
...
+      return toBN(this.totalGasCosts[chainId]);
+    }
+
+    return ethersConstants.Zero; // @todo: Return uint256max in a follow-up change.
```
Comment (james-a-morris) [inline src/clients/ProfitClient.ts:171]:
```suggestion
    return bnZero
```

File: src/dataworker/DataworkerClientHelper.ts [hunk 5]
```diff
@@ -48,7 +48,7 @@ export async function constructDataworkerClients(
   // Disable profitability by default as only the relayer needs it.
   // The dataworker only needs price updates from ProfitClient to calculate bundle volume.
   const profitClient = config.proposerEnabled
-    ? new ProfitClient(logger, commonClients.hubPoolClient, {}, [])
+    ? new ProfitClient(logger, commonClients.hubPoolClient, {}, [], "")
```
Comment (james-a-morris) [inline src/dataworker/DataworkerClientHelper.ts:51]: Resolved offline. This isn't used at all in the dataworker so it's not really needed.

File: src/utils/DepositUtils.ts [hunk 6]
```diff
@@ -117,10 +117,6 @@ export function getUniqueEarlyDepositsInRange(
   );
 }
 
-export function isDepositSpedUp(deposit: Deposit): boolean {
-  return deposit.speedUpSignature !== undefined && deposit.newRelayerFeePct !== undefined;
-}
-
```
Comment (james-a-morris) [inline src/utils/DepositUtils.ts:123]: Moved into the SDK. +1

**Review by james-a-morris — APPROVED** [review]
LGTM - no intended functionality change I've seen here. This is a bump over several patch versions so we should be careful of errors in the rest of the codebase.

File: test/ProfitClient.ConsiderProfitability.ts [hunk 7]
```diff
@@ -99,6 +100,7 @@ describe("ProfitClient: Consider relay profit", () => {
   // Define LOG_IN_TEST for logging to console.
   const { spyLogger }: { spyLogger: winston.Logger } = createSpyLogger();
   let hubPoolClient: MockHubPoolClient, profitClient: MockProfitClient;
+  const message = "0x"; // Empty message by default.
```
[the remark this answers is not in the document]
Comment (pxrl) [reply test/ProfitClient.ConsiderProfitability.ts]: Updated: https://github.com/across-protocol/relayer-v2/pull/1020/commits/ced2d991eb1d0783e5f2257b8579aee45e4986e8

File: src/clients/ProfitClient.ts (hunk 4 above)
Comment (pxrl) [reply src/clients/ProfitClient.ts:172]: Will address this in a follow-up commit :+1:

File: src/clients/ProfitClient.ts (hunk 3 above)
Comment (pxrl) [reply src/clients/ProfitClient.ts]: Updated: https://github.com/across-protocol/relayer-v2/pull/1020/commits/ced2d991eb1d0783e5f2257b8579aee45e4986e8

**Pull request merged**